lib/money/currency.rb

```diff
@@ -0,0 +1,71 @@
+class Money
+  class Currency
+    class UnknownCurrency < ArgumentError; end
+
+    CURRENCY_DATA_PATH = "#{File.expand_path('../../../config', __FILE__)}".freeze
+
+    @@mutex = Mutex.new
+
+    attr_reader :iso_code, :iso_numeric, :name, :smallest_denomination, :subunit_to_unit, :minor_units
+
+    class << self
+      def new(currency_iso)
+        iso = currency_iso.to_s.downcase
+        loaded_currencies[iso] || @@mutex.synchronize { loaded_currencies[iso] = super(iso) }
+      end
+      alias_method :find!, :new
+
+      def find(currency_iso)
+        return nil if currency_iso.blank?
```

Comment:
Remove this and have `raise UnknownCurrency, "Invalid iso4217 currency '#{currency_iso}'"` in `def new`